File: result/translate/starchat/java2python/python/K_TH_LARGEST_SUM_CONTIGUOUS_SUBARRAY.py
def f_gold(arr, n, k):
        sum = [0] * (n + 1)
        sum[0] = 0
        sum[1] = arr[0]
        for i in range(2, n + 1):
            sum[i] = sum[i - 1] + arr[i - 1]
        q = []
        for i in range(1, n + 1):
            for j in range(i, n + 1):
                x = sum[j] - sum[i - 1]
                if len(q) < k:
                    q.append(x)
                else:
                    if q[0] < x:
                        q.pop(0)
                        q.append(x)
        return q[0]

File: result/translate/starchat/java2cpp/c++/K_TH_LARGEST_SUM_CONTIGUOUS_SUBARRAY.cpp
int f_gold ( int arr [ ], int n, int k ) {
    int sum[n+1];
    sum[0] = 0;
    sum[1] = arr[0];
    for(int i=2;i<=n;i++) sum[i] = sum[i-1] + arr[i-1];
    priority_queue<int, vector<int>, greater<int>> pq;
    for(int i=1;i<=n;i++){
        for(int j=i;j<=n;j++){
            int x = sum[j] - sum[i-1];
            if(pq.size() < k) pq.push(x);
            else{
                if(pq.top() < x) pq.pop();
                pq.push(x);
            }
        }
    }
    return pq.top();
}